chains/manager.go

```diff
@@ -900,6 +900,16 @@ func (m *manager) createAvalancheChain(
 		return nil, fmt.Errorf("error initializing network handler: %w", err)
 	}
 
+	nf := &common.NotificationForwarder{
+		Subscribe: linearizableVM.WaitForEvent,
+		Log:       ctx.Log,
+		Engine:    h,
+	}
+
+	cn.OnChange = nf.PreferenceOrStateChanged
+
+	defer nf.Start()
```

Comment:
Similar question about where this is closed

Reply:
Fixed [here](https://github.com/ava-labs/avalanchego/pull/3999/commits/7510692702967fac32c55c5a7d81b04d8b737a49).